### PR TITLE
JC-2189 Use correctly breadcrumbBuilder accessor method for q&a / cr pages containing errors

### PR DESCRIPTION
**Pull request by CocoJumbo**

### CODE REVIEW

**Comment by CocoJumbo** [comment]
I did not notice at once similar pull request https://github.com/jtalks-org/jcommune/pull/87, and created own. Now i entered the course, and I think how to write test.

**Comment by CocoJumbo** [comment]
Writing correct tests for JC-2189

File: jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/CodeReviewControllerTest.java [hunk 1]
```diff
@@ -127,12 +130,21 @@ public void createValidationFail() throws Exception {
...
+        when(breadcrumbBuilder.getNewTopicBreadcrumb(branch)).thenReturn(breadcrumbs);
 
         //invoke the object under test
         ModelAndView mav = controller.createCodeReview(getDto(), result, BRANCH_ID);
 
         //check result
+        List<Breadcrumb> breadcrumbsFromModel = (List) mav.getModel().get("breadcrumbList");
+        for (int i = 0; i < breadcrumbs.size(); i++) {
+            assertEquals(breadcrumbs.get(i).getValue(), breadcrumbsFromModel.get(i).getValue());
+            if(i == 1)
```
Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/CodeReviewControllerTest.java]: Here `if` is redundant. I'd advice take out section bellow from loop

File: jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java [hunk 2]
```diff
@@ -200,13 +202,22 @@ public void createTopicShouldNotPassAndMustShowTopicErrorIfItIsInvalid() throws
...
         ModelAndView mav = controller.createTopic(getDto(), result, BRANCH_ID);
 
         verify(branchService).get(BRANCH_ID);
         verify(breadcrumbBuilder).getNewTopicBreadcrumb(branch);
-        //
+
+        List<Breadcrumb> breadcrumbsFromModel = (List) mav.getModel().get("breadcrumbList");
+        for (int i = 0; i < breadcrumbs.size(); i++) {
+            assertEquals(breadcrumbs.get(i).getValue(), breadcrumbsFromModel.get(i).getValue());
+            if(i == 1)
```
Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java]: Same

File: jcommune-plugins/questions-n-answers-plugin/src/test/java/org/jtalks/jcommune/plugin/questionsandanswers/controller/QuestionsAndAnswersControllerTest.java [hunk 3]
```diff
@@ -158,9 +162,18 @@ public void createQuestionValidationErrorsTest() throws Exception {
 
         when(result.hasErrors()).thenReturn(true);
         when(branchService.get(anyLong())).thenReturn(branch);
+        List<Breadcrumb> breadcrumbs = new BreadcrumbBuilder().getNewTopicBreadcrumb(branch);
+        when(breadcrumbBuilder.getNewTopicBreadcrumb(branch)).thenReturn(breadcrumbs);
 
         String actual = controller.createQuestion(topicDto, result , model, 1L, request);
 
+        for (int i = 1; i < breadcrumbs.size(); i++) {
+            if(i == 1)
```
Comment (mixas27) [inline jcommune-plugins/questions-n-answers-plugin/src/test/java/org/jtalks/jcommune/plugin/questionsandanswers/controller/QuestionsAndAnswersControllerTest.java]: `If` block is redundant here. I'd advice take out section below from loop

File: jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java (hunk 2 above)
Comment (CocoJumbo) [inline jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java]: http://qa.jtalks.org/jcommune/topics/9593

Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java]: I have nothing against copy-paste here. I just want to know why we need theses `if` blocks and asserts inside loop

Comment (mixas27) [inline jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/TopicControllerTest.java]: You should read my comments from down to up :)

**Comment by mixas27** [comment]
Merged

**Pull request closed without merging**